hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncAdmin.java

```diff
@@ -1506,5 +1508,63 @@
    *   The return value will be wrapped by a {@link CompletableFuture}.
    */
   CompletableFuture<Boolean> isSnapshotCleanupEnabled();
+  
+  /**
+   * Gets group info for the given group name
+   * @param groupName the group name
+   * @return group info
+   */
+  CompletableFuture<? extends RSGroupInfo> getRSGroupInfo(String groupName);
+
+  /**
+   * Move given set of servers to the specified target RegionServer group
+   * @param servers set of servers to move
+   * @param targetGroup the group to move servers to
+   */
+  CompletableFuture<Void> moveServers(Set<Address> servers, String targetGroup);
+
+  /**
+   * Creates a new RegionServer group with the given name
+   * @param groupName the name of the group
+   */
+  CompletableFuture<Void> addRSGroup(String groupName);
+
+  /**
+   * Removes RegionServer group associated with the given name
+   * @param groupName the group name
+   */
+  CompletableFuture<Void> removeRSGroup(String groupName);
+
+  /**
+   * Balance regions in the given RegionServer group
+   * @param groupName the group name
+   * @return boolean Whether balance ran or not
+   */
+  CompletableFuture<Boolean> balanceRSGroup(String groupName);
```

Comment:
What is the behavior if there are misplaced regions?

Reply:
They will be reassigned to group servers.